src/lib/StateService.ts

```diff
@@ -0,0 +1,60 @@
+import { BannerMessageState, ConcurrentBuildState, PauseState } from '../db';
+import { config } from './Config';
+
+export class StateService {
+  constructor() {}
+
+  static async pause(reason: string, user: ISessionUser) {
+    await this.unpause();
+    await PauseState.create<PauseState>({
+      pauserAaid: user.aaid,
+      reason,
+    });
+  }
+
+  static async unpause() {
+    await PauseState.truncate();
+  }
+
+  static async getPauseState(): Promise<IPauseState | null> {
+    const state = await PauseState.findOne<PauseState>();
+    return state ? state.get() : null;
+  }
+
+  static async addBannerMessage(
+    message: string,
+    messageType: IMessageState['messageType'],
+    user: ISessionUser,
+  ) {
+    await StateService.removeBannerMessage();
+    await BannerMessageState.create<BannerMessageState>({
+      senderAaid: user.aaid,
+      message,
+      messageType,
+    });
+  }
+
+  static async removeBannerMessage() {
+    await BannerMessageState.truncate();
+  }
+
+  static async getBannerMessageState(): Promise<IMessageState | null> {
+    const state = await BannerMessageState.findOne<BannerMessageState>();
+    return state ? state.get() : null;
+  }
+
+  static async updateMaxConcurrentBuild(maxConcurrentBuilds: number, user: ISessionUser) {
+    await ConcurrentBuildState.truncate();
```

Comment:
I followed the existing patterns for config table. Yeah, we can update all of them in order to maintain history and use the last updated record to get the value of the field. Let me try it out and see the impact of the change.

Reply:
I wouldn't invest too much time doing it for the other tables since they already exist and would therefore need some kind of database table migration. I think its worthwhile doing for this table alone regardless.

Reply:
Agreed, the db migration is not worth it.